crates/sparrow-runtime/src/prepare/preparer.rs

```diff
@@ -185,6 +183,53 @@ impl Preparer {
             self.time_multiplier.as_ref(),
         )
     }
+
+    /// Creates the output url prefix to use for prepared files.
+    ///
+    /// e.g. for osx: file:///<dir>/<KASKADA_PATH>/tables/<table_uuid>/prepared/<uuid>
+    ///      for s3:  s3://<path>/<KASKADA_PATH>/tables/<table_uuid>/prepared/<uuid>
+    pub fn prepared_output_url_prefix(
+        &self,
+        prefix: &ObjectStoreUrl,
+    ) -> error_stack::Result<ObjectStoreUrl, Error> {
+        let error = || Error::InvalidUrl(prefix.to_string());
+
+        // Append the trailing slash to make it a directory
+        let uuid = Uuid::new_v4();
+        let uuid = uuid.to_string() + "/";
+
+        if prefix.is_local() {
+            // The temp directory doesn't have a trailing slash, so we can't join on the existing URL.
+            let prefix = prefix.to_string();
+            let url = prefix
+                + "/"
+                + KASKADA_PATH
+                + "/tables/"
+                + &self.table_config.uuid.to_string()
+                + "/prepared/"
+                + &uuid;
+
+            let url = ObjectStoreUrl::from_str(&url).change_context_lazy(error)?;
+            Ok(url)
+        } else if prefix.is_s3() {
```

Comment:
Why do we need to special case `s3`? If it's not local it's an object store. Why not treat that as general?